contents/tutorials/nextjs-monitoring.md

```diff
@@ -0,0 +1,217 @@
+---
+
+Monitoring a Next.js app ensures the best possible experience for your users. It ensures the optimizations Next.js provides are working as well as possible, and minmizes errors hurting user experience.
+
+## Create a Next.js app
+
+You need to create a Next.js app if you are going to monitor it. To create one, run the following command, choose **not** to use TypeScript, and the defaults for all the other options (including using the app router).
+
+```bash
+npx create-next-app@latest monitor
+```
+
+After installing and creating all the files, go into your newly created `monitor` folder and start the server.
+
+```bash
+cd monitor
+npm run dev
+```
+
+This starts up your Next.js site, which we can now add PostHog to set up monitoring.
+
+![App](../images/tutorials/nextjs-monitoring/app.png)
+
+## Set up PostHog
+
+To add PostHog, install `posthog-js`. 
+
+```bash
+npm i posthog-js
+```
+
+After this, create a `providers.js` file in your `app` folder. In this file, set up PostHog for the client using the `PostHogProvider` component as well as your project API key and instance address, which you can get from the getting started flow or [your project settings](https://app.posthog.com/project/settings).
+
+```js
+// app/provider.js
+'use client'
+import posthog from 'posthog-js'
+import { PostHogProvider } from 'posthog-js/react'
+
+if (typeof window !== 'undefined') {
+  posthog.init('<ph_project_api_key>', {
+    api_host: '<ph_instance_address>'
+  })
+}
+
+export default function PHProvider({ children }) {
+  return <PostHogProvider client={posthog}>{children}</PostHogProvider>
+}
+```
+
+Once created, import and add the provider to your `layout.js` file. 
+
+```js
+// app/layout.js
+import './globals.css'
+import Providers from './providers'
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <Providers>
+        <body>{children}</body>
+      </Providers>
+    </html>
+  )
+}
+```
+
+Once this is set up, go to your [project settings](https://app.posthog.com/project/settings), and make sure:
+
+- Autocapture is enabled
+- Under recordings, record user sessions, capture console logs, and capture network performance are all enabled.
+
+Enabling these provides the key monitoring tools including performance monitoring, app usage, and session replays. It also enables capturing custom events which we use for errors and more performance data next.
+
+## Custom error capture
+
+Another part of monitoring is error capture. There are a couple of ways to set this up with PostHog. First, you could set up and use our [Sentry integration](/docs/libraries/sentry). Second, you could set up custom error capture, which we show here.
+
+To do this, we can create a React error boundary. In the `app` folder, create a file named `error.js`. In this file, set up a basic component that uses `usePostHog` to capture an `$exception` event and returns an error page. It should look like this:
```

Comment:
What is a React error boundary? Perhaps worth linking to react docs somewhere? Or explain it more
```suggestion
To do this, we can create a React error boundary:

1. In the `app` folder, create a file named `error.js`. 
2. In this file, set up a basic component that uses `usePostHog` to capture an `$exception` event and returns an error page. 

It should look like this:
```